### PR TITLE
C API: u8 in Rust corresponds to unsigned char* in C

### PR DESCRIPTION
**Pull request by dannywillems**
# Description

wasm_memory_data returns type is [*mut u8](https://github.com/wasmerio/wasmer/blob/c9e658e25e4a4d915ef2663188360751125a095a/lib/c-api/src/wasm_c_api/externals/memory.rs#L65) which corresponds in C to `unsigned char *`. `byte_t` is used in the C API to represent `*mut u8` but it is an [alias for char](https://github.com/wasmerio/wasmer/blob/c9e658e25e4a4d915ef2663188360751125a095a/lib/c-api/tests/wasm-c-api/include/wasm.h#L37).
When compiling, we get the following warning:
```
warning: pointer targets in initialization of ‘uint8_t *’ {aka ‘unsigned char *’} from ‘byte_t *’ {aka ‘char *’} differ in signedness [-Wpointer-sign]
  572 |    uint8_t* x420 = wasm_memory_data(x419);
```

This MR changes the return type from `byte_t` to `unsigned char`.

### CODE REVIEW

**Review by ptitSeb — APPROVED** [review]

**Comment by ptitSeb** [comment]
Ah yes, good catch.

**Comment by dannywillems** [comment]
I would suggest to add the flags `-Wall -Wextra` in the tests.

**Comment by syrusakbary** [comment]
We can't move from the standard `wasm.h` headers or used types.

If you want to change the `wasm.h` header file, I'd suggest making it a PR to the main `wasm-c-api` repo:
https://github.com/WebAssembly/wasm-c-api/blob/main/include/wasm.h#L478

**Review by syrusakbary — CHANGES_REQUESTED** [review]
We can't merge this as this file is just a mirror of:
https://github.com/WebAssembly/wasm-c-api/blob/main/include/wasm.h#L478

**Comment by dannywillems** [comment]
> We can't merge this as this file is just a mirror of: https://github.com/WebAssembly/wasm-c-api/blob/main/include/wasm.h#L478

See https://github.com/WebAssembly/wasm-c-api/pull/179

**Comment by syrusakbary** [comment]
Closing this as stale, but feel free to reopen the PR once wasm-c-api merges this upstream

**Pull request closed without merging**
